Add backend: class, assigment and fix typescript

chikaharo opened the pull request.

chikaharo committed:
fix typescript, update class and assignment controller backend

hblab-huybd committed:
fix backend controllers and tsconfig

hblab-huybd committed:
Merge remote-tracking branch 'origin/main' into huydinh

hieptuanle committed:
Thêm giao diện demo module Classes

hieptuanle committed:
Demo giao diện announcements

hieptuanle commented on .gitignore:
```diff
@@ -1,4 +1,46 @@
-node_modules
-coverage
+
+# compiled output
```
E merge vào thì nhớ để ý cái cũ có gì nhé @chikaharo

a thấy cái .gitignore e merge vào xong mất cái dòng converage r

hieptuanle commented on tsconfig.json:
```diff
@@ -10,10 +10,10 @@
     "strict": true,
     "noFallthroughCasesInSwitch": true,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Sao lại cần cái này vậy nhỉ? Biến ko được dùng thì xóa đi luôn chứ để lại làm gì?

hieptuanle commented on packages/backend/tsconfig.json:
```diff
@@ -1,14 +1,29 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "target": "ES2022",
```
Sao từ ESNext lại muốn về ES2022 vậy nhỉ?

hieptuanle commented on packages/backend/tsconfig.json:
```diff
@@ -1,14 +1,29 @@
+    "target": "ES2022",
+    "lib": ["ES2022"],
     "baseUrl": "./",
+    "module": "CommonJS",
```
Chuẩn này cũ rồi giờ dùng ESM e nhé

hieptuanle commented on packages/eslint-config/eslint.config.js:
```diff
@@ -1,3 +1,16 @@
 import createConfig from "./create-config.js";
 
-export default createConfig();
+export default {
```
Chưa rõ lý do đổi cấu hình eslint hoàn toàn?

hblab-huybd committed:
reconfig eslint

hieptuanle committed:
Fix some config

hieptuanle committed:
Sửa để pass typecheck và lint, build được

- Sửa backend/tsconfig.json để thêm cấu hình tsc-alias cho phù hợp chuẩn ESM
- Bỏ thư viện postgres-js không dùng, chỉ dùng node-postres (pg)
- Thêm script db:drop để xóa toàn bộ database hiện tại
- Sửa script test để auto drop, drizzle push và chạy test
- Di chuyển thư mục `db` vào trong src, export db để cho các phần cần thiết dùng
- Xóa các file không dùng đến

TODO: Còn phải sửa các test

hieptuanle committed:
Sửa chữa và cập nhật tất cả test cases cho migration Drizzle ORM

- Cập nhật test setup để sử dụng Drizzle thay vì Sequelize
- Fix database connection và lifecycle management trong tests
- Chuyển đổi tất cả model tests sang Drizzle utilities
- Cập nhật auth controller tests với Drizzle schema
- Sửa chữa route integration tests
- Cập nhật helper function tests
- Thêm proper database connection/disconnection cho test environment
- Tất cả 36 test cases đều pass thành công

hieptuanle committed:
Bỏ url không cần thiết trong config/test.json

hieptuanle committed:
Fix eslint

hieptuanle committed:
Merge branch 'main' into huydinh

hieptuanle committed:
Fix eslint

hieptuanle committed:
Thêm pnpm check
